samples/boot/minimal/src/main/java/sample/JWKSetEndpoint.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+import static java.util.Objects.requireNonNull;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.nimbusds.jose.jwk.JWKSet;
+
+public class JWKSetEndpoint extends OncePerRequestFilter {
+
+    static final String WELL_KNOWN_JWK_URIS = "/.well-known/jwk_uris";
+    private final JWKSet jwkSet;
+
+    public JWKSetEndpoint(JWKSet jwkSet) {
+        this.jwkSet = requireNonNull(jwkSet, "jwkSet should not be null");
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+
+        if (ifRequestMatches(request)) {
+            respond(response);
+        } else {
+            filterChain.doFilter(request, response);
+        }
+    }
+
+    protected void respond(HttpServletResponse response) throws IOException {
+        response.setContentType(APPLICATION_JSON_VALUE);
+        try (Writer writer = response.getWriter()) {
+            writer.write(jwkSet.toPublicJWKSet().toJSONObject().toJSONString());
+        }
+    }
+
+    protected boolean ifRequestMatches(HttpServletRequest request) {
+        return request.getRequestURI().endsWith(WELL_KNOWN_JWK_URIS);
```

Comment:
This will also match `/stuff/.well-known/jwk_uris`, right? Is that acceptable or should it be an exact match?

Reply:
Please use `RequestMatcher`, specifically `AntPathRequestMatcher`